### PR TITLE
Add TaskHub support for Redis provider

### PR DESCRIPTION
**Pull request by ConnorMcMahon**
Add basic integration test structure for Redis provider and implement TaskHub relevant methods of RedisOrchestrationService with tests.

### CODE REVIEW

File: Test/DurableTask.Redis.Tests/TestHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        private static RedisTestConfig GetRedisTestConfig()
+        {
+            if (config == null)
+            {
+                config = new RedisTestConfig();
+                IConfigurationRoot root = new ConfigurationBuilder()
+                    .SetBasePath(System.IO.Directory.GetCurrentDirectory())
+                    .AddJsonFile("appsettings.json", optional: true)
```
[the remark this answers is not in the document]
Comment (ConnorMcMahon) [reply Test/DurableTask.Redis.Tests/TestHelpers.cs]: appsettings.json is the standard approach taken for .NET Core. I think I can make app.config work though if it is approved.

[the remark this answers is not in the document]
Comment (cgillum) [reply Test/DurableTask.Redis.Tests/TestHelpers.cs]: Ah, ok. If it's the standard, then I agree let's go with that.

File: src/DurableTask.Redis/RedisOrchestrationService.cs [hunk 2]
```diff
@@ -42,14 +42,13 @@ public RedisOrchestrationService(RedisOrchestrationServiceSettings settings)
...
 
-
         private ConnectionMultiplexer RedisConnection
         {
             get
             {
                 if (this.internalRedisConnection == null)
                 {
-                    throw new InvalidOperationException("The orchestration service is not currently started.");
+                    this.internalRedisConnection = ConnectionMultiplexer.Connect(this.settings.RedisConnectionString);
```
Comment (cgillum) [inline src/DurableTask.Redis/RedisOrchestrationService.cs]: Let's make this async.

File: src/DurableTask.Redis/RedisOrchestrationService.cs [hunk 3]
```diff
@@ -38,6 +42,14 @@ public RedisOrchestrationService(RedisOrchestrationServiceSettings settings)
 
         public int MaxConcurrentTaskActivityWorkItems => this.settings.MaxConcurrentTaskActivityWorkItems;
 
+        private async Task<ConnectionMultiplexer> GetRedisConnection()
```
Comment (cgillum) [inline src/DurableTask.Redis/RedisOrchestrationService.cs]: Per .NET naming design guidelines, the name should end in "Async" i.e. `GetRedisConnectionAsync()`

**Pull request merged**